src/cljs/ataru/virkailija/application/handlers.cljs

```diff
@@ -1179,3 +1189,22 @@
   :application/filter-applications
   (fn [db _]
     (filter-applications db)))
+
+(reg-event-fx
+  :application/navigate-application-list
+  (fn [{:keys [db]} [_ step]]
+    (let [filtered-applications   (-> db :application :filtered-applications)
+          application-count       (count filtered-applications)
+          current-application-key (-> db :application :selected-key)
+          selected-hakukohde      (-> db :application :selected-hakukohde)
+          current-application-idx (util/first-index-of #(= (:key %) current-application-key) filtered-applications)
+          next-application-idx    (if (nil? current-application-idx)
+                                    0
+                                    (+ current-application-idx step))
+          guarded-idx             (cond
```

Comment:
Maku asioita, mut mulle `(mod next-application-idx (count filtered-applications))` viestii selkeemmin, että on kyse tällässees "ympäri juoksevasta" indeksoinnista.

Reply:
Joo, näin oisin halunnut tehdäkin, mutta jostain syystä ajattelin että toi ei ois toiminut oikein negatiivisille arvoille...